src/core/ConfigurableCorePool.ts

```diff
@@ -0,0 +1,126 @@
+import BN from "bn.js";
+import { TickManager } from "../manager/TickManager";
+import { PositionManager } from "../manager/PositionManager";
+import { PoolState } from "../model/PoolState";
+import { Record } from "../entity/Record";
+import { PoolConfig } from "../entity/PoolConfig";
+import { CorePool } from "./CorePool";
+import { ActionType } from "../enum/ActionType";
+import { Serializer } from "../util/Serializer";
+import { Snapshot } from "../entity/Snapshot";
+
+export class ConfigurableCorePool extends CorePool {
+  private poolState: PoolState;
+  private postProcessorCallback: (
+    this: ConfigurableCorePool,
+    recordId: string,
+    actionType: ActionType,
+    actionParams: object
+  ) => void = function (
+    this: ConfigurableCorePool,
+    recordId: string,
+    actionType: ActionType,
+    actionParams: object
+  ) {};
+
+  constructor(poolConfig: PoolConfig);
```

Comment:
Consider using the `optional parameters` for constructor overloading.

Reply:
Sorry, I stand corrected, this wasn't overloading, please ignore my comment.

Reply:
No, you are correct. It's overloading but not what I want actually.